sympy/solvers/ode/ode.py

```diff
@@ -1957,9 +1956,8 @@ def linearity_check(eqs, j, func, is_linear_):
             if matching_hints['no_of_equation'] == 2:
                 if order_eq == 1:
                     type_of_equation = check_linear_2eq_order1(eq, funcs, func_coef)
-                elif order_eq == 2:
-                    type_of_equation = check_linear_2eq_order2(eq, funcs, func_coef)
```

Comment:
Why is `check_linear_2eq_order1` still needed?

Reply:
For linear 2 equations order 1 type 6 and type 7 I think. There were some test cases solved by them. One of them is in the doctests of `dsolve`

Reply:
Is type6 not handled by the current solvers? Does type7 actually work? I thought it just produced the bogus solutions with `x0(t)` in them...

Reply:
I haven't checked for the validity of those solvers yet.

Reply:
That's fine. Let's focus on simplification now